cogs/groupSubjectLinking.py

```diff
@@ -0,0 +1,213 @@
+from functools import reduce
+from typing import Union
+
+from discord import Role, TextChannel, Member, User, Embed
+from discord.ext.commands import Cog, Bot, has_guild_permissions, group, Context, BadArgument
+from discord.ext.tasks import loop
+from discord_components import DiscordComponents
+
+from cogs.botStatus import listener
+from cogs.util.ainit_ctx_mgr import AinitManager
+from core.error.error_collection import GroupOrSubjectNotFoundError, LinkingNotFoundError
+from core.globalEnum import SubjectsOrGroupsEnum, DBKeyWrapperEnum
+from core.logger import get_discord_child_logger
+from core.predicates import bot_chat
+from mongo.study_subject_relation import StudySubjectRelations
+from mongo.subjectsorgroups import SubjectsOrGroups
+
+bot_channels: set[TextChannel] = set()
+first_init = True
+
+logger = get_discord_child_logger("Linking")
+
+
+class Linking(Cog):
+    def __init__(self, bot: Bot):
+        self.bot = bot
+        self.db = StudySubjectRelations(self.bot)
+        self.need_init = True
+        if not first_init:
+            self.ainit.start()
+
+    @listener()
+    async def on_ready(self):
+        global first_init
+        if first_init:
+            first_init = False
+            self.ainit.start()
+
+    @loop()
+    async def ainit(self):
+        """
+        Loads the configuration for the module.
+        """
+        global bot_channels
+        # noinspection PyTypeChecker
+        async with AinitManager(bot=self.bot,
+                                loop=self.ainit,
+                                need_init=self.need_init,
+                                bot_channels=bot_channels) as need_init:
+            if need_init:
+                DiscordComponents(self.bot)
+
+    # link group
+
+    @group(pass_context=True,
+           name="link")
+    @bot_chat(bot_channels)
+    @has_guild_permissions(administrator=True)
+    async def link(self, ctx: Context):
+        if not ctx.invoked_subcommand:
+            raise BadArgument
+        member: Union[Member, User] = ctx.author
+        logger.info(f'User="{member.name}#{member.discriminator}({member.id})", Command="{ctx.message.content}"')
+
+    @link.command(pass_context=True)
+    async def add(self, ctx: Context,
```

Comment:
Procedural question to link roles.
Automatically add links of all higher semesters to make the work easier?